tests/test_schema.py

```diff
@@ -360,15 +363,13 @@ class NestedSchema(Schema):
             bar = fields.Str()
 
             def on_bind_field(self, field_name, field_obj):
-                field_obj.metadata["fname"] = self.context["fname"]
```

Comment:
I modified this test because I couldn't reproduce this with the new context.

I assume this was a very contrived example, not really something we want to do in real life.

This test is now so close to the test above, I'm wondering if we ought to keep it.

Reply:
We could do this:

```py
def test_nested_on_bind_field_hook():
    class MySchema(Schema):
        class NestedSchema(Schema):
            bar = fields.Str()

            def on_bind_field(self, field_name, field_obj):
                field_obj.metadata["fname"] = self.context["fname"]

        foo = fields.Nested(NestedSchema)

    with Context({"fname": "foobar"}):
        schema1 = MySchema()
        assert schema1.fields["foo"].schema.fields["bar"].metadata["fname"] == "foobar"

    with Context({"fname": "quxquux"}):
        schema2 = MySchema()
        assert schema2.fields["foo"].schema.fields["bar"].metadata["fname"] == "quxquux"
```

But it is not the kind of usage I'd like to promote.